pkg/apis/navigator/validation/cassandra_test.go

```diff
@@ -86,3 +93,85 @@ func TestValidateCassandraCluster(t *testing.T) {
 		)
 	}
 }
+
+func TestValidateCassandraClusterUpdate(t *testing.T) {
+	type testT struct {
+		old           *navigator.CassandraCluster
+		new           *navigator.CassandraCluster
+		errorExpected bool
+	}
+
+	setPersistence := func(
+		c *navigator.CassandraCluster,
+		p navigator.PersistenceConfig,
+	) *navigator.CassandraCluster {
+		c = c.DeepCopy()
+		c.Spec.NodePools[0].Persistence = p
+		return c
+	}
+
+	setRack := func(
+		c *navigator.CassandraCluster,
+		rack string,
+	) *navigator.CassandraCluster {
+		c = c.DeepCopy()
+		c.Spec.NodePools[0].Rack = rack
+		return c
+	}
+
+	setDatacenter := func(
+		c *navigator.CassandraCluster,
+		rack string,
+	) *navigator.CassandraCluster {
+		c = c.DeepCopy()
+		c.Spec.NodePools[0].Datacenter = rack
+		return c
+	}
+
+	tests := map[string]testT{
+		"unchanged cluster": {
+			old: validCassCluster,
+			new: validCassCluster,
+		},
+		"changed rack": {
+			old:           validCassCluster,
+			new:           setRack(validCassCluster, "toot"),
+			errorExpected: true,
+		},
+		"changed datacenter": {
+			old:           validCassCluster,
+			new:           setDatacenter(validCassCluster, "doot"),
+			errorExpected: true,
+		},
+		"enable persistence config": {
+			old: setPersistence(validCassCluster, navigator.PersistenceConfig{Enabled: false}),
+			new: validCassCluster,
+		},
+	}
+
+	for title, persistence := range persistenceErrorCases {
+		tests[title] = testT{
+			old:           validCassCluster,
+			new:           setPersistence(validCassCluster, persistence),
+			errorExpected: true,
+		}
+	}
+
+	for title, tc := range tests {
+		t.Run(
+			title,
+			func(t *testing.T) {
+				errs := validation.ValidateCassandraClusterUpdate(tc.old, tc.new)
+				if tc.errorExpected && len(errs) == 0 {
+					t.Errorf("expected error but got none")
+				}
+				if !tc.errorExpected && len(errs) != 0 {
+					t.Errorf("unexpected errors: %s", errs)
+				}
+				for _, e := range errs {
+					t.Logf("error string is: %s", e)
+				}
+			},
+		)
+	}
+}
```

Comment:
:grey_question: tests look good to me, but it gets me thinking whether this might be a good place to use  property based tests (`testing/quick` or gopter). We'd generate a "range" of `CassandraCluster` objects and feed them to the validation function and then assert that if the supplied old and new objects had differences in anything but the whitelisted fields, that a validation error is generated. 
I tried something along these lines in https://github.com/jetstack/navigator/pull/256/commits/490933cab1f77881fce0227abe381a2f99de97d9#diff-37ffe475170e4d658b593d0134efd192R49 and I wouldn't mind discussing the idea sometime.